File: src/main/java/dungeon/warrior/Game.java
package dungeon.warrior;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Game {

    private BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    private boolean astronautsAlive = false;
    private boolean monsterAlive = true;
    private int room = 1;
    private int playerHits = 0;    // Times player has been hit
    private int monsterHits = 0;   // Times monster has been hit
    private boolean hasSword = false;
    private boolean swordAvailable = true;  // Changed to true at start
    private boolean hasMace = false;
    private boolean maceAvailable = true;

    public static void main(String[] args) throws IOException {
        Game game = new Game();
        game.play();
    }

    private void play() throws IOException {
        boolean forever = true;
        String command;

        System.out.println();
        System.out.println("Uvod");
        System.out.println();
        System.out.println("Bio si u svemirskom brodu i provjeravao si motore, kad odjednom je drugi svemirski brod " +
                "napao tvoj svemirski brod. Pokusao si da izadjes iz sobe za motore, ali tacno na izlazu si pao u nesvijest. ");
        System.out.println("Probudio si se. Svemirski brod je bio potpuno unisten.");

        while (forever) {
            if (room == 1) {
                room1();
            } else if (room == 2) {
                room2();
            } else if (room == 3) {
                room3();
            } else if (room == 4) {
                room4();
            }
        }
    }

    private void respawn() {
        System.out.println("Umro si...");
        System.out.println("Probudio si se ponovo na pocetku.");
        room = 1;
        playerHits = 0;    // Only reset hits, keep sword status
    }

    private void room1() throws IOException {
        System.out.println("Nalazis se u prostoriji za motore. Vrata su na zapadu.");

        if (astronautsAlive) {
            System.out.println("Tu se nalaze astronauti u zelenim odijelima. Imaju zelene kacige sa narandzastim staklom.");
        }

        String command = in.readLine();

        if (command.equals("zapad")) {
            room = 2;
        }
    }

    private void room2() throws IOException {
        System.out.println("Nalazis se u hodniku. Prostorija za motore je na istoku. Polu-unistena plava vrata su na jugu.");
        
        if (maceAvailable && !hasMace) {
            System.out.println("U uglu hodnika vidis teški buzdovan!");
        }

        String command = in.readLine();

        if (command.equals("jug")) {
            room = 3;
        } else if (command.equals("istok")) {
            room = 1;
        } else if (command.equals("uzmi buzdovan") && maceAvailable && !hasMace) {
            System.out.println("Uzeo si buzdovan! Ovo je mnogo jače oružje!");
            hasMace = true;
            maceAvailable = false;
        }
    }

    private void room3() throws IOException {
        System.out.println("Nalazis se u sobi koja je zapravo raskrsnica. Osjecas da se nesto trese u blizini. " +
                "Na zapadu su vrata prekrivena krvlju. Na istoku je velika biblioteka. Na sjeveru je hodnik.");
        
        if (swordAvailable && !hasSword) {
            System.out.println("Na podu vidis sjajan mac!");
        }

        String command = in.readLine();

        if (command.equals("zapad")) {
            System.out.println("Vrata ne mogu da se otvore.");
        } else if (command.equals("istok")) {
            room = 4;
        } else if (command.equals("sjever")) {
            room = 2;
        } else if (command.equals("uzmi mac") && swordAvailable && !hasSword) {
            System.out.println("Uzeo si mac! Sada si jaci!");
            hasSword = true;
            swordAvailable = false;
        }
    }

    private void room4() throws IOException {
        System.out.println("Nalazis se u biblioteci. Svijetla trepere. Cujes jako lupanje. Na zapadu je raskrsnica.");
        
        if (monsterAlive) {
            System.out.println("U cosku biblioteke stoji cudoviste. Ima velike kandze i oci koje svijetle u mraku.");
            System.out.println("Cudoviste je primilo " + monsterHits + " udaraca.");
            System.out.println("Ti si primio " + playerHits + " udaraca.");
        }

        String command = in.readLine();

        if (command.equals("zapad")) {
            room = 3;
        } else if (command.equals("napadni") && monsterAlive) {
            if (hasMace) {
                monsterHits += 5;  // Mace deals 5 damage
            } else {
                monsterHits += (hasSword ? 2 : 1); // Sword deals 2, fists deal 1
            }
            
            if (monsterHits >= 10) {
                System.out.println("Pobijedio si cudoviste!");
                monsterAlive = false;
            } else {
                System.out.println(hasMace ? 
                    "Udario si cudoviste buzdovanom! Ono uzvraća udarac!" :
                    hasSword ? 
                        "Udario si cudoviste macem! Ono uzvraća udarac!" :
                        "Udario si cudoviste! Ono uzvraća udarac!");
                playerHits++;
                if (playerHits >= 3) {
                    respawn();
                }
            }
        }
    }

}

File: game.js
const readlineSync = require('readline-sync');

class Game {
	constructor() {
		this.astronautsAlive = false;
		this.monsterAlive = true;
		this.room = 1;
		this.playerHits = 0;
		this.monsterHits = 0;
		this.hasSword = false;
		this.swordAvailable = true;
		this.hasMace = false;
		this.maceAvailable = true;
	}
	
	play() {
		console.log();
		console.log("Uvod");
		console.log();
		console.log("Bio si u svemirskom brodu i provjeravao si motore, kad odjednom je drugi svemirski brod " +
					"napao tvoj svemirski brod. Pokusao si da izadjes iz sobe za motore, ali tacno na izlazu si pao u nesvijest. ");
		console.log("Probudio si se. Svemirski brod je bio potpuno unisten.");
		
		while (true) {
			if (this.room === 1) {
				this.room1();
			} else if (this.room === 2) {
				this.room2();
			} else if (this.room === 3) {
				this.room3();
			} else if (this.room === 4) {
				this.room4();
			}
		}
	}
	
	respawn() {
		console.log("Umro si...");
		console.log("Probudio si se ponovo na pocetku.");
		this.room = 1;
		this.playerHits = 0; // Only reset hits, keep sword status
	}
	
	room1() {
		console.log("Nalazis se u prostoriji za motore. Vrata su na zapadu.");
		if (this.astronautsAlive) {
			console.log("Tu se nalaze astronauti u zelenim odijelima. Imaju zelene kacige sa narandzastim staklom.");
		}
		
		let command = readlineSync.question('');
		
		if (command === "zapad") {
			this.room = 2;
		}
	}
	
	room2() {
		console.log("Nalazis se u hodniku. Prostorija za motore je na istoku. Polu-unistena plava vrata su na jugu.");
		if (this.maceAvailable && !this.hasMace) {
			console.log("U uglu hodnika vidis teški buzdovan!");
		}
		
		let command = readlineSync.question('');
		
		if (command === "jug") {
			this.room = 3;
		} else if (command === "istok") {
			this.room = 1;
		} else if (command === "uzmi buzdovan" && this.maceAvailable && !this.hasMace) {
			console.log("Uzeo si buzdovan! Ovo je mnogo jače oružje!");
			this.hasMace = true;
			this.maceAvailable = false;
		}
	}
	
	room3() {
		console.log("Nalazis se u sobi koja je zapravo raskrsnica. Osjecas da se nesto trese u blizini. " +
					"Na zapadu su vrata prekrivena krvlju. Na istoku je velika biblioteka. Na sjeveru je hodnik.");
		if (this.swordAvailable && !this.hasSword) {
			console.log("Na podu vidis sjajan mac!");
		}
		
		let command = readlineSync.question('');
		
		if (command === "zapad") {
			console.log("Vrata ne mogu da se otvore.");
		} else if (command === "istok") {
			this.room = 4;
		} else if (command === "sjever") {
			this.room = 2;
		} else if (command === "uzmi mac" && this.swordAvailable && !this.hasSword) {
			console.log("Uzeo si mac! Sada si jaci!");
			this.hasSword = true;
			this.swordAvailable = false;
		}
	}
	
	room4() {
		console.log("Nalazis se u biblioteci. Svijetla trepere. Cujes jako lupanje. Na zapadu je raskrsnica.");
		if (this.monsterAlive) {
			console.log("U cosku biblioteke stoji cudoviste. Ima velike kandze i oci koje svijetle u mraku.");
			console.log("Cudoviste je primilo " + this.monsterHits + " udaraca.");
			console.log("Ti si primio " + this.playerHits + " udaraca.");
		}
		
		let command = readlineSync.question('');
		
		if (command === "zapad") {
			this.room = 3;
		} else if (command === "napadni" && this.monsterAlive) {
			if (this.hasMace) {
				this.monsterHits += 5;
			} else {
				this.monsterHits += (this.hasSword ? 2 : 1);
			}
			
			if (this.monsterHits >= 10) {
				console.log("Pobijedio si cudoviste!");
				console.log("Zavrsio si Dungeon Warrior! Aj zdravo!");
				process.exit(0);
				this.monsterAlive = false;
			} else {
				console.log(this.hasMace ? 
					"Udario si cudoviste buzdovanom! Ono uzvraća udarac!" :
					this.hasSword ? 
						"Udario si cudoviste macem! Ono uzvraća udarac!" :
						"Udario si cudoviste! Ono uzvraća udarac!");
				this.playerHits++;
				if (this.playerHits >= 3) {
					this.respawn();
				}
			}
		}
	}
}

const game = new Game();
game.play();
